src/builtins/core/duration.rs

```diff
@@ -77,7 +77,7 @@ impl PartialDuration {
 /// `Duration` is made up of a `DateDuration` and `TimeDuration` as primarily
 /// defined by Abtract Operation 7.5.1-5.
 #[non_exhaustive]
-#[derive(Debug, Clone, Copy, Default)]
+#[derive(Debug, Clone, Copy, Default, PartialEq, PartialOrd)]
```

Comment:
issue: `Duration` should not implement `PartialOrd` and potentially not `PartialEq`.

The `compare` method supersedes the `cmp` impl from `PartialOrd`, implementing them both could lead to confusion. `PartialEq` is not necessarily needed by the specification. It doesn't define an `equals` method on `Duration`, but `PartialEq` is incredibly useful for testing purposes.